### DISCUSSION TITLE
How to rotate ObjectBucketClaim's user credentials

### DISCUSSION THREAD

**Discussion by ibotty** [rook/rook#14760] [General]
I'd like to be able to rotate the credentials of the owner of a object bucket claim.  How do I do that?  Maybe using `radosgw-admin user modify`?

**Comment by BlaineEXE** [comment]
`radosgw-admin` is one way to do so, yes. The other way would be to use RGW's admin ops (REST) API: https://docs.ceph.com/en/latest/radosgw/adminops/

`radosgw-admin` is likely easier, but for some types of automation, REST can be preferable for users.

[Addendum]

Restart the `rook-ceph-operator*` pod by deleting it to force Rook to reprocess all OBCs.

S3 buckets can have several credentials active at one time. Rook will always use the first credentials in the list (i.e., at index `[0]`) when updating secret info, so the credentials that you want OBC applications to use should be first in the list.

**Reply by ibotty** [reply]
Thank you for the answer.  Will it be reflected in secret that is managed by rook when changed using bare-bones ceph/radosgw?

**Reply by BlaineEXE** [reply]
Right! I forgot to mention that in my first reply. Restart the `rook-ceph-operator*` pod by deleting it to force Rook to reprocess all OBCs. 

S3 buckets can have several credentials active at one time. Rook will always use the first credentials in the list (i.e., at index `[0]`) when updating secret info, so the credentials that you want OBC applications to use should be first in the list.

**Reply by ibotty** [reply]
That worked like a charm:
```
> oc rsh -n rook-ceph deploy/rook-ceph-tools radosgw-admin user modify --uid obc-XXXXXX --access-key XXXXXX --gen-secret
> oc rollout restart -n rook-ceph deploy/rook-ceph-operator
```
and after waiting a bit the new secret access key was used.

Thank you!
